makeit/gengodep

```diff
@@ -11,27 +37,63 @@ Please remove the build directory ("builddir" by default) and run
 mconfig again.
 ========================================================================
 
-	EOT
-	exit 1
-fi
+EOT
+    exit 1
+  fi
+}
 
-shift
+function go_dependency_template() {
+  cat <<'TEMPLATE'
+    {{ with $d := . }}
+      {{ if not $d.Standard }}
+        {{ range $d.GoFiles }}
+          {{ printf "%s/%s\n" $d.Dir . }}
+        {{ end }}
+        {{ range $d.CgoFiles }}
+          {{ printf "%s/%s\n" $d.Dir . }}
+        {{ end }}
+      {{ end }}
+    {{ end }}
+TEMPLATE
+}
 
-go=$1
-var=$2
-gotags=$3
-depfile=$4
+function find_embedded_files () {
+  local -r module_file="${1}"
+  grep "go:embed" "${module_file}" | sed -e 's|.* ||'
+}
 
-shift 4
+function report_embedded_files() {
+  local -r module_file="${1}"
+  local module_directory
+  module_directory="$(dirname "${module_file}")"
+  for embedded_file in $(find_embedded_files "${module_file}" ); do
+    printf '%s/%s\n' "${module_directory}" "${embedded_file}"
+  done
+}
 
-# this is needed so that the value we are getting from the makefile does
-# get propagated down to go list.
-export GOPROXY
+function output_deps_file() {
+  local -r go="${1}"
+  local -r source_prefix="${2}"
+  local -r gotags="${3}"
+  local -r depfile="${4}"
 
-template='{{ with $d := . }}{{ if not $d.Standard }}{{ range $d.GoFiles }}{{ printf "%s/%s\n" $d.Dir . }}{{ end }}{{ range $d.CgoFiles }}{{ printf "%s/%s\n" $d.Dir . }}{{ end }}{{ end }}{{ end }}'
+  shift 4
+  local template dependency_list
+  template="$(go_dependency_template)"
+  echo "$@"
+  dependency_list=$( (${go} list -deps -f "${template}" -tags "${gotags}" "$@" ||true) | sort -u)
```

Comment:
With `errexit` set, it's probably safer to remove `|| true` here to fail and not return an empty dependency list